File: 250723/약수가 세 개인 수/numbers-with-three-divisors.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.util.StringTokenizer

fun main() {
    val (start, end) = BufferedReader(System.`in`.reader()).use {
        val st = StringTokenizer(it.readLine())
        Pair(st.nextToken().toInt(), st.nextToken().toInt())
    }
    BufferedWriter(System.out.writer()).use {
        val result = (start..end).filter { it ->
            (1..it).filter { i -> it % i == 0 }.size == 3
        }.size
        it.write(result.toString())
    }
}

File: 250722/약수가 세 개인 수/numbers-with-three-divisors.cpp
#include <iostream>
using namespace std;

int main() {
    int start, end, result = 0;
    cin >> start >> end;
    for (int i = start; i <= end; i++) {
        int cnt = 0;
        for (int j = 1; j <= i; j++) {
            if (i % j == 0) {
                cnt++;
            }
        }
        if (cnt == 3) {
            result++;
        }
    }
    cout << result;
    return 0;
}
